### DISCUSSION TITLE
Qodana Action use local IDE instead of Docker images

### DISCUSSION THREAD

**Discussion by shyim** [JetBrains/Qodana#191] [Ideas]
Hey,

I just saw that qodana-cli supports local IDE with `--ide PATH`. I would prefer to use a local IDE in GitHub Actions, as you can modify the environment in a much nicer way.

Let's say you need a specific node version for your NPM stuff. Just use `setup-node` action or PHP with a specific version just `setup-php` action. It feels more like integrated with GitHub Actions as you have the same environment in your Qodana Run.

**Comment by brichbash** [comment]
Hi,

You're right. It's an experimental feature, but you can already try it in [the nightly version](https://github.com/JetBrains/qodana-cli/releases/tag/nightly) on GitHub Actions, an example usage can be seen [here](https://github.com/JetBrains/qodana-action/blob/32c3156f63f31a54dece4ce3e32dbb767c38bb61/.github/workflows/ci.yml#L142). The full support is planned by the end of this year.
